src/conv_transpose2d.js

```diff
@@ -0,0 +1,180 @@
+'use strict';
+
+import {Tensor} from './lib/tensor.js';
+import {validateConv2dParams} from './lib/validate-input.js';
+import {computePaddingForAutoPad} from './lib/compute-padding.js';
+import {transpose} from './transpose.js';
+
+/**
+ * Compute a 2-D transposed convolution given 4-D input and filter tensors.
+ * @param {Tensor} input
+ * @param {Tensor} filter
+ * @param {MLConvTranspose2dOptions} options
+ * @return {Tensor}
+ */
+export function convTranspose2d(input, filter, {padding = [0, 0, 0, 0],
+  strides = [1, 1],
+  groups = 1,
+  dilations = [1, 1],
+  outputPadding = [0, 0],
+  outputSizes,
+  activation = (x) => x,
+  inputLayout = 'nchw',
+  filterLayout = 'iohw',
+  bias,
+  autoPad = 'explicit',
+}
+= {}) {
+  if (inputLayout === 'nhwc') {
+    // nhwc -> nchw
+    input = transpose(input, {permutation: [0, 3, 1, 2]});
+  }
+  if (filterLayout === 'iohw') {
+    // iohw -> oihw
+    filter = transpose(filter, {permutation: [1, 0, 2, 3]});
```

Comment:
Hmm, I'm confused by the filter nomenclature (granted, ConvTranspose is one of my least familiar operators) because the default is "iohw" in the spec, but the code is transposing it to "oihw", which isn't an [enum in the list](https://www.w3.org/TR/webnn/#api-mlgraphbuilder-convtranspose2d). I'm not saying it should be changed in either this code or in the spec, just curious that the algorithm below isn't coded for the default filter layout, given that is presumably the more common one found in models. Is it more efficient to do so?

- [ONNX uses](https://github.com/onnx/onnx/blob/main/docs/Operators.md#convtranspose) filter shape order "iohw" (C x M/group x kH x kW).
- [TF uses](https://www.tensorflow.org/api_docs/python/tf/nn/conv2d) filter shape order "hwio" `[filter_height, filter_width, in_channels, out_channels]`.

Reply:
Ah, your newer comment explains that you're reusing the logic for Conv2d, and hence you're switching it to "oihw" to be compatible with that. 👍